aspnetcore/fundamentals/http-context.md

```diff
@@ -129,7 +129,7 @@ public class UserRepository : IUserRepository
 
 ## HttpContext access from a background thread
 
-`HttpContext` isn't thread-safe. Reading or writing properties of the `HttpContext` outside of processing a request can result in a <xref:System.NullReferenceException>.
+`HttpContext` isn't thread-safe. Reading or writing properties of the `HttpContext` outside of processing a request can result in a <xref:System.NullReferenceException>. In most cases, code should not access <xref:Microsoft.AspNetCore.Http.IHttpContextAccessor.HttpContext?displayProperty=nameWithType> to get data. Rather than accessing `IHttpContextAccessor`, applications should copy the data from the request on the controller or middleware and then pass it to a service.
```

Comment:
I'm not sure if this is the right place for this change. The opening line from this topic suggests using `IHttpContextAccessor` with services, but then this argues against it. I think it's reasonable to reference `IHttpContextAccessor` from a service, but the complication is if said service does background work.

I wonder if this might discourage the reader from using `IHttpContextAccessor` inside services that don't perform background work, which could mean they'll end up copying data and trying to pass it through when it's not necessary to do so.

Reply:
Agreed.

Reply:
```suggestion
`HttpContext` isn't thread-safe. Reading or writing properties of the `HttpContext` outside of processing a request can result in a <xref:System.NullReferenceException>.
```